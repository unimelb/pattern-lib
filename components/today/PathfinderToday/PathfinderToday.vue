<template>
  <div
    :class="{
      'pathfinder-today': true,
      'pathfinder-today--dark': dark,
    }">
    <div class="max">
      <slot name="before" />
      <div role="navigation">
        <ul class="pathfinder-today__list grid grid--lg">
          <li
            v-for="(item, index) in items"
            :key="index"
            class="pathfinder-today__list-item cell cell--lg cell--tab-1of2 cell--wide-1of4">
            <a
              :id="item.id"
              :href="item.href"
              class="pathfinder-today__link">
              <span
                :class="{
                  'pathfinder-today__link-title': true,
                  'heading-sm': true,
                  'heading-sm--link': true,
                  'pathfinder-today__link-title--two-lines': titleTwoLines,
                }">{{ item.title }}</span>
              <span class="pathfinder-today__link-icon">
                <SvgIcon
                  name="arrow-right"
                  aria-hidden="true" />
              </span>
            </a>
            <span class="pathfinder-today__link-description label-s">{{
              item.description
            }}</span>
          </li>
        </ul>
      </div>
      <slot name="after" />
    </div>
  </div>
</template>

<script>
import SvgIcon from 'components/icons/SvgIcon.vue';

export default {
  components: {
    SvgIcon,
  },
  props: {
    items: {
      type: Array,
      required: true,
      default() {
        return [];
      },
    },
    inline: {
      type: Boolean,
      required: false,
      default: false,
    },
    dark: {
      type: Boolean,
      required: false,
      default: false,
    },
    titleTwoLines: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
};
</script>
