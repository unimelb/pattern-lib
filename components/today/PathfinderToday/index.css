.pathfinder-today {
  background: rgb(var(--col-grey-light));
  padding-top: 2.5rem;
  padding-bottom: 2.5rem;

  @media (--bp-tablet) {
    padding-top: 3.5rem;
    padding-bottom: 3.5rem;
  }

  &--dark {
    background: none;
    color: rgb(var(--col-white));
  }

  &--dark &__link-icon {
    background-color: rgb(var(--col-blue-mid-2));
    color: rgb(var(--col-white));
  }

  &--dark &__link {
    border-top: 1px solid rgb(var(--col-white), .2);

    &:hover {
      .pathfinder-today__link-title {
        color: rgb(var(--col-white));
      }
    }
  }

  &--dark &__link-title,
  &--dark &__link-description {
    color: rgb(var(--col-white));
  }

  &__list {
    list-style-type: none;
    padding: 0;
    margin-bottom: 0;

    &-item {
      margin: 0;
    }
  }

  &__link {
    text-decoration: none;
    display: flex;
    flex-direction: column;
    height: 100%;
    border-top: 1px solid rgb(var(--col-grey-mid-3), .2);
    padding-top: 1rem;
    padding-bottom: 1rem;
    position: relative;

    @media (--bp-tablet) {
      padding-top: 1.5rem;
      padding-bottom: 1.5rem;
    }

    @media (--bp-wide) {
      padding-top: 1rem;
      padding-bottom: 0;
    }

    &:hover {
      background-color: none;

      .pathfinder-today__link-title {
        color: rgb(var(--col-bg-tertiary-navy));
      }

      .pathfinder-today__link-icon {
        background-color: rgb(var(--col-bg-tertiary-navy));
      }
    }

    &-title {
      color: rgb(var(--col-navy-dark));
      padding-right: 2.5rem;
      display: block;
      transition: color .15s ease;

      @media (--bp-tablet) {
        max-width: 12rem;
      }
    }

    &-icon {
      position: absolute;
      top: .8rem;
      right: 0;
      width: 2rem;
      height: 2rem;
      display: flex;
      align-items: center;
      justify-content: center;
      background-color: rgb(var(--col-navy-dark));
      color: rgb(var(--col-white));
      border-radius: 50%;
      transition: background .15s ease;

      svg {
        width: .75rem;
        height: .75rem;
      }

      @media (--bp-tablet) {
        top: 1.3rem;
      }

      @media (--bp-wide) {
        top: .8rem;
      }
    }

    &-description {
      color: rgb(var(--col-grey-mid-2));
      font-weight: normal;
      display: block;
      margin-bottom: 0;
      margin-top: 1rem;
      max-width: 15rem;

      @media (--bp-max-mobile) {
        display: none;
      }
    }
  }
}
