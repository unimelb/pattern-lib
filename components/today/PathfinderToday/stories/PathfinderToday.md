## PathfinderToday


| props  	| type    	|
|--------	|---------	|
| items  	| array   	|
| inline 	| boolean 	|
| dark   	| boolean 	|

| slot  	| type    	|
|--------	|---------	|
| after  	| html   	|

**Note**: Inside of `<pathfinder-today>` component the `items` property is an array of pathfinder with `title`, `description`, and `href`:

```json
pathfinderItems: [
  {
    title: 'Undergraduate courses',
    description: 'Flexible first degrees that you can tailor to your career goals, or explore new possibilities.',
    href: '/landing-page/',
  },
  {
    title: 'Graduate courses',
    description: 'Study in small classes led by experts alongside others who share your interests and goals.',
    href: '/landing-page/',
  },
  {
    title: 'Graduate research',
    description: 'Access dedicated research centres, advanced  facilities, generous scholarships and more.',
    href: '/landing-page/',
  },
  {
    title: 'Short courses and microcredentials',
    description: 'Upskill, expand your knowledge and take the next step in your career with focused courses.',
    href: '/landing-page/',
  },
]
```

Rules: 

- Min/Max of 4 items
- Title (100 characters)
- Description (200 characters)

```html
<pathfinder-today items="pathfinderItems">
  <slot />
</pathfinder-today>
```