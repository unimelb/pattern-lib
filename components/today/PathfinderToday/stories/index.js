import { storiesOf } from '@storybook/vue';
import { createStory } from 'storybook/utils';

import Story1 from './Story1.vue';
import Story2 from './Story2.vue';
import Story3 from './Story3.vue';
/* ##Import story component here */

storiesOf('Today/Pathfinder', module)
  .add('Default', createStory(Story1))
  .add('Inline Dark', createStory(Story2))
  .add('Slot', createStory(Story3));
/* ##Story goes here */
