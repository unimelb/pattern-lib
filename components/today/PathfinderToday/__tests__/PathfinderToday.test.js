import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import PathfinderToday from '../PathfinderToday.vue';

expect.extend(toHaveNoViolations);

const pathfinderLinks = [
  {
    title: 'Undergraduate courses',
    description: 'Flexible first degrees that you can tailor to your career goals, or explore new possibilities.',
    href: '/landing-page/',
  },
  {
    title: 'Graduate courses',
    description: 'Study in small classes led by experts alongside others who share your interests and goals.',
    href: '/landing-page/',
  },
  {
    title: 'Graduate research',
    description: 'Access dedicated research centres, advanced  facilities, generous scholarships and more.',
    href: '/landing-page/',
  },
  {
    title: 'Short courses and microcredentials',
    description: 'Upskill, expand your knowledge and take the next step in your career with focused courses.',
    href: '/landing-page/',
  },
];

describe('PathfinderToday', () => {
  it('should match snapshot', () => {
    const result = shallow(PathfinderToday, {
      propsData: {
        items: pathfinderLinks,
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(PathfinderToday, {
      propsData: {
        items: pathfinderLinks,
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
