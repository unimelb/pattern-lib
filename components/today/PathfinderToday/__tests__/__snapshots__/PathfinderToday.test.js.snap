// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PathfinderToday should match snapshot 1`] = `
<div
  class="pathfinder-today"
>
  <div
    class="max"
  >
     
    <div
      role="navigation"
    >
      <ul
        class="pathfinder-today__list grid grid--lg"
      >
        <li
          class="pathfinder-today__list-item cell cell--lg cell--tab-1of2 cell--wide-1of4"
        >
          <div
            class="pathfinder-today__link"
          >
            <a
              href="/landing-page/"
            >
              <span
                class="pathfinder-today__link-title heading-sm heading-sm--link"
              >
                Undergraduate courses
              </span>
               
              <span
                class="pathfinder-today__link-icon"
              >
                <!---->
              </span>
            </a>
             
            <span
              class="pathfinder-today__link-description label-s"
            >
              Flexible first degrees that you can tailor to your career goals, or explore new possibilities.
            </span>
          </div>
        </li>
        <li
          class="pathfinder-today__list-item cell cell--lg cell--tab-1of2 cell--wide-1of4"
        >
          <div
            class="pathfinder-today__link"
          >
            <a
              href="/landing-page/"
            >
              <span
                class="pathfinder-today__link-title heading-sm heading-sm--link"
              >
                Graduate courses
              </span>
               
              <span
                class="pathfinder-today__link-icon"
              >
                <!---->
              </span>
            </a>
             
            <span
              class="pathfinder-today__link-description label-s"
            >
              Study in small classes led by experts alongside others who share your interests and goals.
            </span>
          </div>
        </li>
        <li
          class="pathfinder-today__list-item cell cell--lg cell--tab-1of2 cell--wide-1of4"
        >
          <div
            class="pathfinder-today__link"
          >
            <a
              href="/landing-page/"
            >
              <span
                class="pathfinder-today__link-title heading-sm heading-sm--link"
              >
                Graduate research
              </span>
               
              <span
                class="pathfinder-today__link-icon"
              >
                <!---->
              </span>
            </a>
             
            <span
              class="pathfinder-today__link-description label-s"
            >
              Access dedicated research centres, advanced  facilities, generous scholarships and more.
            </span>
          </div>
        </li>
        <li
          class="pathfinder-today__list-item cell cell--lg cell--tab-1of2 cell--wide-1of4"
        >
          <div
            class="pathfinder-today__link"
          >
            <a
              href="/landing-page/"
            >
              <span
                class="pathfinder-today__link-title heading-sm heading-sm--link"
              >
                Short courses and microcredentials
              </span>
               
              <span
                class="pathfinder-today__link-icon"
              >
                <!---->
              </span>
            </a>
             
            <span
              class="pathfinder-today__link-description label-s"
            >
              Upskill, expand your knowledge and take the next step in your career with focused courses.
            </span>
          </div>
        </li>
      </ul>
    </div>
     
  </div>
</div>
`;
